### ISSUE TITLE
Changing env_logger's log format

### ISSUE THREAD

**Issue by nicksanders** [rust-lang-nursery/log#47]
What's the recommended way to change env_logger's log format?

I have tried to override the Log trait for Logger to change the log format but can't see how to do it.

**Comment by sfackler** [comment]
env_logger's output format is hard coded at the moment: https://github.com/rust-lang/log/blob/master/env/src/lib.rs#L178-L182

**Comment by nicksanders** [comment]
I have had a go at implementing a way to set a custom log format

https://github.com/nicksanders/log/commit/098f9b71c51078c33b1c260ca04ee25ed8898cd2

I have added a function init_with_log_writer (see below for an example). If its along the right lines let me know and I can add tests and documentation as needed.

```rust

#[macro_use]
extern crate log;
extern crate env_logger;
extern crate time;

use log::{LogLevel, LogRecord};
use env_logger::LogWriter;

struct CustomLogWriter;

impl LogWriter for CustomLogWriter {
    fn write(&self, record: &LogRecord) {
        let t = time::now();
        println!(
            "{},{:03} - {} - {}",
            time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
            t.tm_nsec / 1000_000,
            record.level(),
            record.args()
        );
    }
}

fn main() {
    env_logger::init_with_log_writer(CustomLogWriter).unwrap();
    error!("this is printed by default");
}
```

**Comment by nicksanders** [comment]
I have had a another go, this time using a closure for formatting the log.

It also allows you to change the default log level and environmental variable.

I think its a lot nicer, what do you think?

https://github.com/nicksanders/log/commit/381f44e902fcd68aa6b57a2f80287b1b1c0e67b1

```rust

#[macro_use]
extern crate log;
extern crate env_logger;
extern crate time;

use log::{LogRecord, LogLevelFilter};
use env_logger::LogConfig;

fn main() {
    let formatter = |record: &LogRecord| {
        let t = time::now();
        format!("{},{:03} - {} - {}",
            time::strftime("%Y-%m-%d %H:%M:%S", &t).unwrap(),
            t.tm_nsec / 1000_000,
            record.level(),
            record.args()
        )
    };

    let config = LogConfig {
        env_var: "RUST_LOGGER",
        format: Box::new(formatter),
        level: LogLevelFilter::Info,
    };

    env_logger::init_with_config(config).unwrap();

    error!("error message");
    info!("info message");
}
```

**Comment by softprops** [comment]
+1

**Comment by flying-sheep** [comment]
can we get this in? i’d really like to see the time without wrapping all the macros with my own…

**Comment by nicksanders** [comment]
This has just been merged and released :)

See #48

**Comment by flying-sheep** [comment]
haha awesome! my story with rust so far:

started translating some python script on thursday, got it done monday (only three days!).

met 3 bugs/feature requests that i required, all of which are fixed by now: this one, alexcrichton/flate2-rs#30 (which i pinned down and the author fixed), shepmaster/sxd-document#27 (which i fixed, but it isn’t merged yet)
